.github/workflows/testing.yml

```diff
@@ -0,0 +1,37 @@
+name: Testing
+
+env:
+  pythonDefaultValue: '3.8'
+  clangppDefaultValue: '6.0'
+
+on:
+  pull_request:
+    branches: [ develop, master ]
+  push:
+    branches: [ develop, master ]
+    paths-ignore:
+      - 'licenses/**'
+      - 'LICENSE.md'
+      - 'README.md'
+      - 'RELEASE-NOTES.txt'
+
+jobs:
+  run-tests:
+    name: run stan tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [env.pythonDefaultValue]
+      fail-fast: false
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v2
+        with:
+          submodules:  true
+      - name: Install clang++
+        run: sudo apt-get install clang-${{ env.clangppDefaultValue }}++
+      - name: Run tests
```

Comment:
Is this necessary?

```bash
echo "CXX=clang-6.0" >> make/local
echo "O=0" >> make/local
```

Reply:
Looking at the log, it looks like not.
```
sudo apt-get install clang-6.0++
  shell: /bin/bash -e {0}
  env:
    pythonDefaultValue: 3.8
    clangppDefaultValue: 6.0
Reading package lists...
Building dependency tree...
Reading state information...
clang-6.0 is already the newest version (1:6.0-1ubuntu2).
The following NEW packages will be installed:
  clang-6.0-doc clang-6.0-examples libclang-6.0-dev python-clang-6.0
0 upgraded, 4 newly installed, 0 to remove and 13 not upgraded.
```
I'm not sure if we need/make use of any in `clang-6.0-doc clang-6.0-examples libclang-6.0-dev python-clang-6.0`

Reply:
O=0 is there to make the tests run faster and eat up less RAM (Stan tests with models are known to take up a ton of RAM).

`CXX=clang-6.0` may not be necessary, we just need to convince tests to use clang 6 some way.